### ISSUE TITLE
Null AlternateText on the Image Model fails accessibility check

### ISSUE THREAD

**Issue by RyanDurkin** [sdl/dxa-web-application-dotnet#57]
I'm building a website where accessibility is key and one of the issues we've come across is the rendering of the image alt attribute. If that field is null then MVC strips out the tag

View:
```
<img src="@Model.Image.Url" alt="@Model.Image.AlternateText">
```
Rendered:
```
<img src="/images/image.jpg">
```

This unfortunately is invalid for accessibility see http://webaim.org/techniques/alttext/#basics

> Every image must have an alt attribute. This is a requirement of HTML standard (with perhaps a few exceptions in HTML5). Images without an alt attribute are likely inaccessible. In some cases, images may be given an empty or null alt attribute (e.g., alt="").

If you change the Image Entity Model in Sdl.Web.Tridion.Tests.Models.Entity then an empty alt attribute will be rendered which will pass validation e.g.

```
private string _alternativeText;

[SemanticProperty("s:name")]
[SemanticProperty("c:altText")]
public string AlternateText
{
    get => !string.IsNullOrEmpty(_alternativeText) ? _alternativeText : string.Empty;
    set => _alternativeText = value;
}
```

**Comment by bkoopman** [comment]
I think this is a highly debatable point, I don't consider an empty alt tag more accessible than no alt tag, so the accessibility failure should rather be fixed by supplying an alternate text for the image if you ask me.

**Comment by RyanDurkin** [comment]
Your reply was my first thought too but this isn't a highly debatable point if you read the standards for accessibility. It hasn't got anything to do with our considerations, it's going on the standards set by the accessibility community.
Here's another link for you: https://dequeuniversity.com/rules/axe/1.1/image-alt 
> Ensure all informative <img> elements have short, descriptive alternate text and all decorative <img> elements have empty alt attributes

**Comment by bkoopman** [comment]
I have seen the standards, but the debatable part is how much support for the standard you want built in, since as soon as you supply an alternate text, we support the standard, there is no requirement in the standard to have empty alt attributes, so you can just make sure there is content for them ;)

**Comment by RyanDurkin** [comment]
> there is no requirement in the standard to have empty alt attributes

I just don't get what you're saying there? I've supplied two links stating that they must have empty alt attributes?

**Comment by bkoopman** [comment]
That is only if there is no content. What I'm saying is: just make sure there is content, because that is actually what accessibility is all about. When there is content, you don't need empty attributes.

**Comment by RyanDurkin** [comment]
"Just make sure there is content" is a ridiculous reply. I can't believe I'm having to fight to get this simple change in. Here's another link for you and then I guess I'll give up: https://www.w3.org/TR/WCAG20-TECHS/H67.html

**Comment by NunoLinhares** [comment]
I have to side with @RyanDurkin on this one - looks like it's 1) a small change and 2) correct from the standards compliance point of view (i.e., tags for which there is no ALT content should have it empty rather than absent)

**Comment by DominicCronin** [comment]
Just to make it a bit more concrete: imagine you are a blind person using a screen reader and it keeps reading out stuff like "top left decorative corner".

**Comment by jhorsman** [comment]
@DominicCronin Good point.
Let's stick with the standards indeed.

I guess that <img src="image.jpg"> means: we did not think about an alt text. I wonder what the screen reader would do with this tag, maybe it does read it out with the filename from the search or so.
And then <img src="image.jpg" alt=""> means: this is a decorative image, it does not contain content. So don't read it out at all.

**Comment by bkoopman** [comment]
https://jira.sdl.com/browse/TSI-2891
